server/src/main/java/org/elasticsearch/index/IndexNotFoundException.java

```diff
@@ -37,7 +37,7 @@ public IndexNotFoundException(String index) {
     }
 
     public IndexNotFoundException(String index, Throwable cause) {
-        super("no such index", cause);
+        super("no such index [" + index + "]", cause);
```

Comment:
there is another ctor with a custom message `public IndexNotFoundException(String message, String index)`, I wonder shall we revisit custom messages and include index name there as well for the sake of consistency.

Reply:
I agree, the other ctor should be addressed, so that all constructions of this exception include the index name in the message.

Reply:
@vladimirdolzhenko @rjernst Shouldn't the custom messages be turned into more specific exceptions? Or more specific constructors?

Reply:
@atapin i don't think that it is worth to have a more specific exception and/or more specific ctor than existed one:  https://github.com/elastic/elasticsearch/blob/237650e9c054149fd08213b38a81a3666c1868e5/server/src/main/java/org/elasticsearch/index/IndexNotFoundException.java#L30 

do you have some specific case in mind ?

Reply:
@vladimirdolzhenko I would better suggest to always encaplsulate `no such index [indexName]` and append custom message to that default message. So the ctor would look like: 
```java
public IndexNotFoundException(String message, String index) { 
    super("no such index [" + index + "] and " + message)
}
```

Reply:
sounds great, could you please adjust messages at appropriate call sites ?

Reply:
@vladimirdolzhenko yes sure